localstack/utils/bootstrap.py

```diff
@@ -376,6 +379,113 @@ def extract_port_flags(user_flags, port_mappings: PortMappings):
     return user_flags
 
 
+class ContainerConfigurators:
+    """
+    A set of useful container configurators that are typical for starting the localstack container.
+    """
+
+    @staticmethod
+    def mount_docker_socket(cfg: ContainerConfiguration):
+        source = config.DOCKER_SOCK
+        target = "/var/run/docker.sock"
+        if cfg.volumes.find_target_mapping(target):
+            return
+        cfg.volumes.add(VolumeBind(source, target))
+        cfg.env_vars["DOCKER_HOST"] = f"unix://{target}"
+
+    @staticmethod
+    def mount_localstack_volume(host_path: str | os.PathLike = None):
+        host_path = host_path or config.VOLUME_DIR
+
+        def _cfg(cfg: ContainerConfiguration):
+            if cfg.volumes.find_target_mapping(constants.DEFAULT_VOLUME_DIR):
+                return
+            cfg.volumes.add(VolumeBind(str(host_path), constants.DEFAULT_VOLUME_DIR))
+
+        return _cfg
+
+    @staticmethod
+    def random_gateway_port(cfg: ContainerConfiguration):
+        cfg.ports.add(get_free_tcp_port(), constants.DEFAULT_PORT_EDGE)
+
+    @staticmethod
+    def default_gateway_port(cfg: ContainerConfiguration):
+        cfg.ports.add(constants.DEFAULT_PORT_EDGE)
+
+    @staticmethod
+    def random_container_name(cfg: ContainerConfiguration):
+        name = f"localstack-{short_uid()}"
+        cfg.name = name
+        cfg.env_vars["MAIN_CONTAINER_NAME"] = cfg.name
+
+    @staticmethod
+    def default_container_name(cfg: ContainerConfiguration):
+        cfg.name = config.MAIN_CONTAINER_NAME
+        cfg.env_vars["MAIN_CONTAINER_NAME"] = cfg.name
+
+    @staticmethod
+    def service_port_range(cfg: ContainerConfiguration):
+        cfg.ports.add([config.EXTERNAL_SERVICE_PORTS_START, config.EXTERNAL_SERVICE_PORTS_END])
+
+    @staticmethod
+    def debug(cfg: ContainerConfiguration):
+        cfg.env_vars["DEBUG"] = "1"
+
+    @staticmethod
+    def network(network: str):
+        def _cfg(cfg: ContainerConfiguration):
+            cfg.network = network
+
+        return _cfg
+
+
+def get_gateway_port(container: Container) -> int:
```

Comment:
While I agree with the sentiment - in this case this would only happen if we manually manufacture the container like this, right? All the configurators should prevent this from happening. The impact will be minimal, until we use a similar utility for a container not under our complete control.

I think it is perfectly fine to just raise a ValueError in this case, since the users would be us, not the enduser.

Reply:
thanks both for your comments. i quickly implemented simon's suggestion and added a test to document the behavior